crates/hulk/src/webots/keyboard_device.rs

```diff
@@ -29,14 +28,28 @@ impl KeyboardDevice {
             }
         };
 
-        let chest_button_pressed = received_message.is_some()
-            || key == Some(Keyboard::CONTROL | Keyboard::SHIFT | 'C' as u32);
+        let (control_shift_c_pressed, control_shift_x_pressed, control_shift_u_pressed) =
+            if let Some(key) = self.keyboard.get_key() {
+                const CONTROL_SHIFT_MASK: u32 = Keyboard::CONTROL | Keyboard::SHIFT;
+                const CONTROL_SHIFT_C_MASK: u32 = CONTROL_SHIFT_MASK | 'C' as u32;
+                const CONTROL_SHIFT_X_MASK: u32 = CONTROL_SHIFT_MASK | 'X' as u32;
+                const CONTROL_SHIFT_U_MASK: u32 = CONTROL_SHIFT_MASK | 'U' as u32;
+                (
+                    key & CONTROL_SHIFT_C_MASK == CONTROL_SHIFT_C_MASK,
+                    key & CONTROL_SHIFT_X_MASK == CONTROL_SHIFT_X_MASK,
+                    key & CONTROL_SHIFT_U_MASK == CONTROL_SHIFT_U_MASK,
+                )
+            } else {
+                (false, false, false)
+            };
```

Comment:
I'm afraid this logic is still broken. Pressing Ctrl + Shift + W for example triggers both the chest button and all three head buttons.

I would suggest simply checking for equality like before. I think that way is more sound anyways, you shouldn't just ignore modifier keys you don't require. If the user presses Mod + Ctrl + Shift + C, that probably shouldn't trigger our keybindings.
If you _do_ want to keep bitschubsing, I'd suggest using the `Keyboard::KEY` mask for removing the modifier bits before comparing to the numerical character value.

Reply:
Yeah, since Webots does not support pressing multiple keys at once, this whole feature does not make sense anymore. I'm now only checking whether the `key` is equal to the `*_MASK` and not ANDing it before. This should get near to where we came from.